CMakeLists.txt

```
@@ -222,19 +222,21 @@ endif()
 # We want Kokkos to be built with C++14, since that's what we're using in
 # Parthenon.
 set(CMAKE_CXX_STANDARD 14)
-if (PARTHENON_IMPORT_KOKKOS)
-  find_package(Kokkos 3)
-  if (NOT Kokkos_FOUND)
-    unset(PARTHENON_IMPORT_KOKKOS CACHE)
-    message(FATAL_ERROR "Could not find external Kokkos. Consider importing a Kokkos installation into your environment or disabling external Kokkos with e.g. -DPARTHENON_IMPORT_KOKKOS=OFF")
-  endif()
-else()
-  if (EXISTS ${Kokkos_ROOT}/CMakeLists.txt)
-    add_subdirectory(${Kokkos_ROOT} Kokkos)
-  elseif(EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/external/Kokkos/CMakeLists.txt)
-    add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/external/Kokkos Kokkos)
+if (NOT TARGET Kokkos::kokkos)
+  if (PARTHENON_IMPORT_KOKKOS)
+    find_package(Kokkos 3)
+    if (NOT Kokkos_FOUND)
+      unset(PARTHENON_IMPORT_KOKKOS CACHE)
+      message(FATAL_ERROR "Could not find external Kokkos. Consider importing a Kokkos installation into your environment or disabling external Kokkos with e.g. -DPARTHENON_IMPORT_KOKKOS=OFF")
+    endif()
   else()
-    message(FATAL_ERROR "Could not find Kokkos source. Consider running `git submodule update --init`, providing the path to a Kokkos source directory with Kokkos_ROOT, or setting PARTHENON_IMPORT_KOKKOS=ON to link to an external Kokkos installation.")
+    if (EXISTS ${Kokkos_ROOT}/CMakeLists.txt)
+      add_subdirectory(${Kokkos_ROOT} Kokkos)
```

```suggestion
      add_subdirectory(${Kokkos_ROOT} Kokkos)
      message(STATUS "Using Kokkos source from Kokkos_ROOT=${Kokkos_ROOT}")
```

It just occurred to me that these status messages may be helpful in cases where the enviroment/module/spack system already provides Kokkos (potentially without the user knowing).

---

Good suggestion!